src/ows/src/main/java/org/geoserver/ows/Dispatcher.java

```diff
@@ -1530,6 +1530,14 @@ void parseKVP(Request req) throws ServiceException {
     }
 
     Map<String, Object> parseKVP(Request req, Map<String, Object> kvp) {
+        // if the service is not explicitly set, some WFS parsers will not be found
+        if (!kvp.containsKey("service")
+                && !kvp.containsKey("SERVICE")
```

Comment:
The KVP map shoud be a CaseInsensitivemap. Little known fact, all keys in OGC services are case insensitive, one can write a valid request with "sErViCe=WFS". However, values are case sensitive, so "service=wfs" is actually not valid, strictly speaking)